frontend/src/features/properties/components/forms/subforms/PagedBuildingForms.tsx

```diff
@@ -91,10 +122,9 @@ const PagedBuildingForms: React.FC<PagedBuildingFormsProps> = (props: PagedBuild
                     variant="link"
                     title="Remove Building"
                     onClick={() => {
-                      if (currentPage === buildings.length - 1) {
-                        setCurrentPage(currentPage - 1);
-                      }
-                      arrayHelpers.remove(currentPage);
+                      formikProps.dirty
```

Comment:
Does this work as intended? To me this means:
If the entire form has been modified from the initial state, show the delete dialog. Otherwise just delete the building.

I think it would make more sense if it:
Checked to see if the building being deleted has any non-default content, if so, show the delete dialog when delete is clicked
- or -
Check to see if the building being deleted has an ID (meaning it exists in the DB) if it does, show the delete dialog.

Reply:
Hmm yes you're correct, going to implement a different solution as you suggested